clang-tools-extra/clang-tidy/ClangTidyDiagnosticConsumer.cpp

```diff
@@ -562,9 +567,10 @@ void ClangTidyDiagnosticConsumer::checkFilters(SourceLocation Location,
   }
 
   StringRef FileName(File->getName());
-  LastErrorRelatesToUserCode = LastErrorRelatesToUserCode ||
-                               Sources.isInMainFile(Location) ||
-                               getHeaderFilter()->match(FileName);
+  LastErrorRelatesToUserCode =
+      LastErrorRelatesToUserCode || Sources.isInMainFile(Location) ||
+      (getHeaderFilter()->match(FileName) &&
+       (ExcludeHeaderFilter ? !ExcludeHeaderFilter->match(FileName) : true));
```

Comment:
`ExcludeHeaderFilter ? !ExcludeHeaderFilter->match(FileName) : true` -> `!(ExcludeHeaderFilter  && ExcludeHeaderFilter->match(FileName))`

Reply:
Thank you! Simplified that snippet and added the tests as suggested.